org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/IFilter.java

```diff
@@ -28,12 +30,17 @@
 	 *            class name
 	 * @param superClassName
 	 *            superclass name
+	 * @param classAnnotations
+	 *            annotations qualified name on class
+	 * @param sourceFileName
+	 *            file source name
 	 * @param methodNode
 	 *            method to inspect
 	 * @param output
 	 *            callback to report filtering results to
 	 */
-	void filter(String className, String superClassName, MethodNode methodNode,
+	void filter(String className, String superClassName,
```

Comment:
Now we're definitely beyond the limit of extra parameters. Probably we will see even more in future. I would propose to introduce a new interface

    public interface org.jacoco.core.internal.analysis.filter.IFilterContext{
        String getClassName();
        String getSuperClassName();
        Set<String> getClassAnnotations();
        String getSourceFileName();
    }

For now this can be directly implemented by `ClassAnalyzer`.

Reply:
I thought about introducing something like `IFilterContext` but decided that it's a bit more structural change than this PR needs.